Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -411,7 +411,119 @@ private void RunMatchExpression(FlexMatch binary64DecodedMatch, AnalyzeContext c
+                    {
+                        // The validator determined the match is a false positive.
+                        // i.e., it is not the kind of artifact we're looking for.
+                        // We should suspend processing and move to the next match.
+                        return;
+                    }
+
+                    case Validation.None:
+                    case Validation.ValidatorReturnedIllegalValue:
+                    {
+                        // The validator returned a bad value.
+                        // TODO: we should log this condition
+                        // and then continue processing.
+                        level = FailureLevel.Error;
+                        break;
+                    }
+
+                    case Validation.Authorized:
+                    {
+                        level = FailureLevel.Error;
+                        break;
+                    }
+
+                    case Validation.Unauthorized:
+                    {
+                        level = FailureLevel.Warning;
+                        break;
+                    }
+
+                    case Validation.Expired:
+                    {
+                        level = FailureLevel.Warning;
+                        break;
+                    }
+
+                    case Validation.HostUnknown:
+                    {
+                        level = FailureLevel.Warning;
+                        break;
+                    }
+
+                    case Validation.InvalidForConsultedAuthorities:
+                    {
+                        level = FailureLevel.Warning;
+                        break;
+                    }
+
+                    case Validation.Unknown:
+                    {
+                        level = FailureLevel.Warning;
+                        break;
+                    }
+
+                    case Validation.ValidatorNotFound:
+                    {
+                        break;
+                    }
+
+                    default:
+                    {
+                        throw new InvalidOperationException($"Unrecognized validation value '{state}'.");
+                    }
+                }
+            }
+
+            IList<string> arguments = GetMessageArguments(
+                _argumentNameToIndex,
+                context.TargetUri.LocalPath,
+                base64Encoded: false,
+                matchExpression.MessageArguments);
+
+            Result result = this.ConstructResult(
+                    context.TargetUri,
+                    reportingDescriptor.Id,
+                    level,
+                    region: null,
+                    flexMatch: null,
+                    null,
+                    matchExpression.Fixes,
+                    arguments);
+
+            context.Logger.Log(reportingDescriptor, result);
+        }
+
+        private string NormalizeValidator(string validatorMessage)
```

Comment:
>NormalizeValidator [](start = 23, length = 18)

This should be named 'NormalizedValidatorMessage',my mistake from previous change.

Reply:
Ok, will rename :)

---
In reply to: [554470118](https://github.com/microsoft/sarif-pattern-matcher/pull/96#discussion_r554470118) [](ancestors = 554470118)